addons/danger/functions/fnc_leaderStaticDeploy.sqf

```diff
@@ -65,39 +66,46 @@ private _EH = _gunner addEventHandler ["WeaponAssembled", {
 [formationLeader _gunner, "aware", "AssembleThatWeapon"] call FUNC(doCallout);
 
 // find position ~ kept simple for now!
-private _weaponPos = [ getpos _gunner, 0, 15, 2, 0, 0.19, 0, [], [getpos (leader _gunner), getpos (leader _gunner)]] call BIS_fnc_findSafePos;
-_weaponPos set [2, 0];
+if (_weaponPos isEqualTo []) then {
+
+    _weaponPos = [ getpos _gunner, 0, 15, 2, 0, 0.19, 0, [], [getpos _assistant, getpos _assistant]] call BIS_fnc_findSafePos;
+    _weaponPos set [2, 0];
+
+};
 
 // ready units
 {
     doStop _x;
     _x setUnitPosWeak "MIDDLE";
+    _x forceSpeed 24;
     _x setVariable [QGVAR(forceMove), true];
     _x setVariable [QGVAR(currentTask), "Deploy Static Weapon"];
     _x doMove _weaponPos;
-    //_x setDestination [_weaponPos, "LEADER DIRECT", false];
 
 } foreach [_gunner, _assistant];
 
 // do it
 [
     {
         // condition
-        params ["_gunner", "_assistant", "_pos", "_weaponPos", ""];
-        (unitReady _gunner || {_gunner distance2d _assistant < 3}) || {fleeing _gunner} || {fleeing _assistant}
+        params ["_gunner", "_assistant", "", "_weaponPos", ""];
+        (_gunner distance2d _weaponPos < 2 || {_gunner distance2d _assistant < 3}) || {fleeing _gunner} || {fleeing _assistant}
+        // use of OR here to facilitiate the sometimes shoddy irreverent A3 pathfinding ~ nkenny
     },
     {
         // on near gunner
-        params ["_gunner", "_assistant", "_pos", ""];
+        params ["_gunner", "_assistant", "_pos", "", ""];
```

Comment:
Empty strings in params area only needed if you want to skip an entry that is before a desired one. Empty string at the end make no sense as you can just leave them out and then it is the same behavior.

Reply:
Fixed